applicationset/generators/merge.go

```diff
@@ -221,6 +224,28 @@ func (m *MergeGenerator) GetRequeueAfter(appSetGenerator *argoprojiov1alpha1.App
 
 }
 
+func getMergeGenerator(r argoprojiov1alpha1.ApplicationSetNestedGenerator) *argoprojiov1alpha1.MergeGenerator {
+	if r.Merge == nil {
+		return nil
+	}
+	merge, err := argoprojiov1alpha1.ToNestedMergeGenerator(r.Merge)
+	if err != nil {
+		return nil
+	}
+	return merge.ToMergeGenerator()
+}
+
+func getMatrixGenerator(r argoprojiov1alpha1.ApplicationSetNestedGenerator) *argoprojiov1alpha1.MatrixGenerator {
+	if r.Matrix == nil {
+		return nil
+	}
+	matrix, err := argoprojiov1alpha1.ToNestedMatrixGenerator(r.Matrix)
+	if err != nil {
+		return nil
+	}
+	return matrix.ToMatrixGenerator()
+}
+
```

Comment:
Now that we have created this new functions, can we remove the duplicate code and call this functions there?
Places I saw similar code [here](https://github.com/argoproj/argo-cd/blob/master/applicationset/generators/merge.go#L142-L161) and [here](https://github.com/argoproj/argo-cd/blob/master/applicationset/generators/matrix.go#L83-L105).